packages/frontend/web-ui/src/auth/pages/RegisterConfirm.jsx

```diff
@@ -1,54 +1,23 @@
 import { Link as RouterLink } from 'react-router-dom';
-import { useDispatch, useSelector } from 'react-redux';
-import { Alert, Grid, Link, Typography } from '@mui/material';
+import { Alert, AlertTitle, Button, Grid, Link } from '@mui/material';
 import { RegisterConfirmLayout } from '../layout/RegisterConfirmLayout';
-import { useEffect, useState } from 'react';
 import { CheckingAuth } from '../components/CheckingAuth';
-import { httpClient } from '../../common/http/services/HttpService';
-import { buildSerializableResponse } from '../../common/http/helpers/buildSerializableResponse';
-import { createAuthByToken } from '../../app/store/features/authSlice';
-
-const CODE_QUERY_PARAM = 'code';
-const STATUS_FULFILLED = 'fulfilled';
-const STATUS_IDLE = 'idle';
-const STATUS_PENDING = 'pending';
+import {
+  STATUS_FULFILLED,
+  STATUS_PENDING,
+  STATUS_REJECTED,
+  useRegisterConfirm,
+} from '../hooks/useRegisterConfirm';
 
 export const RegisterConfirm = () => {
-  const [isHidden, setIsHidden] = useState(false);
-
-  const [status, setStatus] = useState(STATUS_IDLE);
-
-  const url = new URL(window.location.href);
-  const codeParam = url.searchParams.get(CODE_QUERY_PARAM);
-
-  const dispatch = useDispatch();
-  const { token, errorMessage } = useSelector((state) => state.auth);
+  const { status, errorMessage } = useRegisterConfirm();
 
-  useEffect(() => {
-    if (status === STATUS_PENDING && token !== null) {
-      updateUserMe(token);
-      setStatus(STATUS_FULFILLED);
-      setIsHidden(true);
-    }
-  }, [token, status, errorMessage]);
-
-  window.onload = async () => {
-    setStatus(STATUS_PENDING);
-    if (codeParam !== null) {
-      await dispatch(createAuthByToken(codeParam));
-    }
+  const isRegisterConfirmOk = () => {
```

Comment:
This method has a different beahvior than the one suggested by its name. A better name should be found

Reply:
Done, I changed it for **getRegisterConfirmOkDisplay**